### PR TITLE
KOGITO-3589 Extract apps from runtimes

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-3589

Dependent PRs:
- https://github.com/kiegroup/kogito-pipelines/pull/89
- https://github.com/kiegroup/kogito-runtimes/pull/909
- https://github.com/kiegroup/kogito-apps/pull/554

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by radtriste** [comment]
Jenkins retest this

**Comment by radtriste** [comment]
testing in progress

File: Jenkinsfile.deploy [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+}
+
+void commitAndCreatePR(String repo) {
+    def commitMsg = "[${getBuildBranch()}] Update project version to ${getProjectVersion()} for release"
+    def prBody = "Generated by build ${BUILD_TAG}: ${BUILD_URL}.\nPlease do not merge, it will be merged automatically after testing."
+    // Not using githubscm.commitChanges() because globbing won't work.
+    // See: https://github.com/kiegroup/kogito-runtimes/pull/570#discussion_r449268738
+    dir(repo) {
+        addNotIgnoredPoms()
+        sh "git commit -m '${commitMsg}'"
```
Comment (Kevin-Mok) [inline Jenkinsfile.deploy]:
```suggestion
    dir(repo) {
        githubscm.commitChanges(commitMsg, { githubscm.findAndStageNotIgnoredFiles('pom.xml') })
```

File: Jenkinsfile.deploy [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+        if ! git check-ignore -q "\$pom"; then
+            poms_to_add="\$poms_to_add \$pom"
+        fi
+    done < found_poms.txt
+    rm found_poms.txt
+    git add \$poms_to_add
+    git status
+    '''
+}
+
```
Comment (Kevin-Mok) [inline Jenkinsfile.deploy]:
```suggestion
```

File: Jenkinsfile.promote [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+    while IFS= read -r pom; do
+        if ! git check-ignore -q "\$pom"; then
+            poms_to_add="\$poms_to_add \$pom"
+        fi
+    done < found_poms.txt
+    rm found_poms.txt
+    git add \$poms_to_add
+    '''
+}
+
```
Comment (Kevin-Mok) [inline Jenkinsfile.promote]:
```suggestion
```

File: Jenkinsfile.promote [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+void commitAndCreatePR(String repo) {
+    def commitMsg = "[${getBuildBranch()}] Update snapshot version to ${getSnapshotVersion()}"
+    def prBody = "Generated by build ${BUILD_TAG}: ${BUILD_URL}.\nPlease do not merge, it will be merged automatically after testing."
+    // Not using githubscm.commitChanges() because globbing won't work.
+    // See: https://github.com/kiegroup/kogito-runtimes/pull/570#discussion_r449268738
+    addNotIgnoredPoms()
+    sh "git commit -m '${commitMsg}'"
```
Comment (Kevin-Mok) [inline Jenkinsfile.promote]:
```suggestion
    
    githubscm.commitChanges(commitMsg, { githubscm.findAndStageNotIgnoredFiles('pom.xml') })
```

**Review by Kevin-Mok — CHANGES_REQUESTED** [review]

File: Jenkinsfile.promote [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+}
+
+void installGithubCLI() {
+    sh """
+    wget https://github.com/cli/cli/releases/download/v${env.GITHUB_CLI_VERSION}/gh_${env.GITHUB_CLI_VERSION}_linux_amd64.tar.gz
+    tar xzf gh_${env.GITHUB_CLI_VERSION}_linux_amd64.tar.gz
+    mv gh_${env.GITHUB_CLI_VERSION}_linux_amd64/bin/gh .
+    rm -r gh_${env.GITHUB_CLI_VERSION}_linux_amd64*
+    """
+}
```
Comment (rsynek) [inline Jenkinsfile.promote:258]: Maybe worth moving to the shared libraries? (not in scope for this PR)

**Review by rsynek — APPROVED** [review]

File: Jenkinsfile.promote (hunk 5 above)
Comment (radtriste) [reply Jenkinsfile.promote:258]: will do in a future PR ;)

**Comment by radtriste** [comment]
Testing ok https://docs.google.com/spreadsheets/d/1L4NvDuQ6h4nOVuXMgpVmlvF-sjoTH6eI6EJFLWiwe7I/edit?usp=sharing
@rsynek good to merge

**Pull request merged**